vulcanai2/models/basenetwork.py

```diff
@@ -168,40 +187,56 @@ def forward(self, inputs, **kwargs):
         if not isinstance(inputs, list):
             inputs = [inputs]
 
-        if self.input_networks is not None:
+        if self.input_networks:
             net_outs = []
-            for net, x in zip(self.input_networks, inputs):
-                net_outs.append(net(x))
+            for in_net, x in zip(self.input_networks.values(), inputs):
```

Comment:
could you add a comment describing the concept here? took me a while to parse.

Reply:
sure